tests/Integration/DatabaseMonitoringTest.php

```diff
@@ -57,6 +58,67 @@ public function testInjection()
         ]);
     }
 
+    public function testInjectionServiceMappingOnce()
+    {
+        try {
+            $hook = \DDTrace\install_hook(self::class . "::instrumented", function (HookData $hook) {
+                $span = $hook->span();
+                Integration::handleInternalSpanServiceName($span, "pdo");
+                $span->name = "instrumented";
+                DatabaseIntegrationHelper::injectDatabaseIntegrationData($hook, 'mysql', 1);
+            });
+            self::putEnv("DD_TRACE_DEBUG_PRNG_SEED=42");
+            self::putEnv("DD_DBM_PROPAGATION_MODE=full");
```

Comment:
Would it be useful to have a test with partial as well? As we do have the service in partial mode IIRC.

Reply:
Added the tests for `none` & `service`; I couldn't find [documentation](https://datadoghq.atlassian.net/wiki/spaces/TS/pages/3183772859/Troubleshooting+DBM+and+APM+Correlation) (or [public doc](https://docs.datadoghq.com/database_monitoring/connect_dbm_and_apm/?tab=go)) for `service`, though